FLT/Mathlib/LinearAlgebra/Span/Defs.lean

```diff
@@ -0,0 +1,28 @@
+import Mathlib.LinearAlgebra.Span.Defs
+
+open Pointwise
+
+variable {R : Type*} [Semiring R]
+
+variable {M : Type*} [AddCommMonoid M] [Module R M]
+
+variable {p p' : Submodule R M}
+
+variable {P : M → Prop}
+
+namespace Submodule
+
+@[simp high]
+lemma forall_mem_sup : (∀ x ∈ p ⊔ p', P x) ↔ (∀ x₁ ∈ p, ∀ x₂ ∈ p', P (x₁ + x₂)) := by
+  simp [mem_sup]
+  aesop
```

Comment:
Here is a (rare) situation where a non-terminal simp is OK. Because aesop uses simp (it uses the kitchen sink) so if a change in simp breaks this proof then it was probably going to break anyway. After a non-terminal simp you can only have very strong tactics, basically.